contracts/amm.cairo

```diff
@@ -88,23 +77,36 @@ func do_trade{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
     strike_price: Math64x61_,
     maturity: Int,
     side: OptionSide,
-    option_size: Math64x61_,
+    option_size: Int,
     quote_token_address: Address,
     base_token_address: Address,
     lptoken_address: Address
 ) -> (premia: Math64x61_) {
-    // options_size is always denominated in base tokens (ETH in case of ETH/USDC)
+    // options_size is always denominated in the lowest possible unit of base tokens (ETH in case of ETH/USDC), e.g. wei in case of ETH.
+    // Option size of 1 ETH would be 10**18 since 1 ETH = 10**18 wei.
+    // Option size of 100 USD would be 100*10**6 since USDC is divisible up to 6 decimal places.
 
     alloc_locals;
 
     with_attr error_message("do_trade premia calculation and updates failed") {
         // 0) Helper values
-        let (option_size_in_pool_currency) = convert_amount_to_option_currency_from_base(
-            option_size,
+        with_attr error_message("conversions failed in do_trade"){
+            let option_size_uint256 = intToUint256(option_size);
+            let strike_price_uint256 = toUint256_balance(strike_price, quote_token_address);
+        }
+        let (option_size_in_pool_currency: Uint256) = convert_amount_to_option_currency_from_base_uint256(
+            option_size_uint256,
             option_type,
-            strike_price
+            strike_price_uint256,
+            base_token_address
         );
 
+        let option_size_m64x61_ = Math64x61.fromFelt(option_size);
+        let (base_decimals: felt) = IERC20.decimals(contract_address=base_token_address);
+        let (base_div) = pow10(base_decimals);
+        let base_div_m64x61 = Math64x61.fromFelt(base_div);
+        let option_size_m64x61 = Math64x61.div(option_size_m64x61_, base_div_m64x61);
```

Comment:
we have shitload of different convertor functions and none fits this case? for example this one `fromUint256_balance `???

Reply:
fixed with fromInt_balance... probably coded this before I had fromInt